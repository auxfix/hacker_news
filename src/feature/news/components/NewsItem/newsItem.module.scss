@import '~styles/global.scss';
.container {
    border: 1px solid $main_color
}