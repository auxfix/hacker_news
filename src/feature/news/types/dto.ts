export interface HackerStoryDTO {
    id: number;
    title: string;
    url: string;
    timestamp: number;
    score: number;
    by: string;
  }

export interface UserDTO {
    id: string;
    created: number;
    karma: number;
    about: string;
  }