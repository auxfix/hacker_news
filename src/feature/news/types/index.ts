export interface hackerstory {
  authorid: string;
  id: number;
  img?: string;
  karma?: number;
  score: number;
  time: number;
  title: string;
  url: string;
}

export interface user {
  about: string;
  created: number;
  id: string;
  karma: number;
}
