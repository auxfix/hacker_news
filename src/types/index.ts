export * from './common';
export * from './state';
export * from 'feature/news/types';
