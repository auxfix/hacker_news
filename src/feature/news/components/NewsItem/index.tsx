import { HackerStory } from "types";
import styles from './newsItem.module.scss';

interface NewsProps {
    newsItem: HackerStory;
}

export default function NewsItem(props: NewsProps){
    const { newsItem } = props;
    return (
     <div className={styles.container}key={newsItem.id}>
        <div>{newsItem.id}</div>
        <div>{newsItem.title}</div>
        <div>{newsItem.url}</div>
        <div>{newsItem.score}</div>
        <div>{newsItem.timestamp}</div>
    </div>
   )
}