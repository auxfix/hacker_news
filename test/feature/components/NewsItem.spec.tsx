import React from 'react';
import NewsItem from 'feature/news/components/NewsItem';
import { render, screen } from 'test-utils';
import { HackerStory } from 'types';

describe('NewsItem', () => {
  const testNewsItem: HackerStory = {
    authorId: 'hacker.one',
    id: 242342343,
    score: 74,
    time: 1669726441,
    title: 'amazing article',
    url: 'https://en.wikipedia.org/wiki/Hacker',
    img: 'https://images/nice_image.jpg',
    karma: 34,
  };

  it('should render properly', () => {
    render(<NewsItem newsItem={testNewsItem} />);

    expect(screen.getByTestId('NewsItem')).toMatchSnapshot();
    expect(screen.getByTestId('score')).toHaveTextContent(testNewsItem.score.toString());
    expect(screen.getByTestId('author')).toHaveTextContent(testNewsItem.authorId);
    expect(screen.getByTestId('title')).toHaveTextContent(testNewsItem.title);
    expect(screen.getByTestId('karma')).toHaveTextContent(testNewsItem.karma?.toString());
    expect(screen.getByTestId('link')).toHaveAttribute('href', testNewsItem.url);
  });
});
