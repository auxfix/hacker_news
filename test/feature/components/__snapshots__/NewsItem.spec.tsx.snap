// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should render properly 1`] = `
<div
  data-testid="NewsItem"
>
  <img
    alt="News cover"
    src="https://images/nice_image.jpg"
  />
  <div>
    <div>
      <span
        data-testid="day"
      >
        Tuesday
      </span>
      <span
        data-testid="monthDateYear"
      >
        November 29 2022
      </span>
    </div>
    <h1
      data-testid="title"
    >
      amazing article
    </h1>
    <div>
      <div>
        <span>
          Story score:
        </span>
        <span
          data-testid="score"
        >
          74
        </span>
      </div>
      <div>
        <span>
          Author:
        </span>
        <span
          data-testid="author"
        >
          hacker.one
        </span>
      </div>
      <div>
        <span>
          Author Karma:
        </span>
        <span
          data-testid="karma"
        >
          34
        </span>
      </div>
    </div>
    <a
      data-testid="link"
      href="https://en.wikipedia.org/wiki/Hacker"
      rel="noreferrer"
      target="_blank"
    >
      Read more
    </a>
  </div>
</div>
`;
